archive/imo/imo2011_q5.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2021 Alain Verberkmoes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alain Verberkmoes
+-/
+
+import tactic
+
+/-!
+# IMO 2011 Q5
+
+Let `f` be a function from the set of integers to the set
+of positive integers.  Suppose that, for any two integers
+`m` and `n`, the difference `f(m) − f(n)` is divisible by
+`f(m − n)`.  Prove that, for all integers `m` and `n` with
+`f(m) ≤ f(n)`, the number `f(n)` is divisible by `f(m)`.
+-/
+
+open int
+
+theorem imo2011_q5 (f : ℤ → ℤ) (hpos : ∀ n : ℤ, 0 < f n)
+  (hdvd : ∀ m n : ℤ, f (m - n) ∣ f m - f n) :
+∀ m n : ℤ, f m ≤ f n → f m ∣ f n :=
+begin
+  intros m n h_fm_le_fn,
+  cases lt_or_eq_of_le h_fm_le_fn with h_fm_lt_fn h_fm_eq_fn,
+  { -- m < n
+    let d := f m - f (m - n),
+    have h_fn_dvd_d : f n ∣ d,
+    { rw ←sub_sub_self m n,
+      exact hdvd m (m - n) },
+    have h_d_lt_fn : d < f n,
+    { refine lt_trans _ h_fm_lt_fn,
+      exact sub_lt_self _ (hpos (m - n)) },
```

Comment:
While backwards reasoning proofs certainly work for inequalities, presenting the argument as a `calc` block is much more readable:
```suggestion
    { calc d < f m : sub_lt_self _ (hpos (m - n)) 
         ... < f n : h_fm_lt_fn, },
```

Reply:
(To be clear, I'm not suggesting this is a necessary change. I just mention it as you're new to mathlib PRs and I wasn't sure you knew about `calc`. There is an argument that proofs in `archive` should be "exemplars", so if you believe it is better style then probably it's worth doing. :-)